src/post.jsx

```diff
@@ -1,40 +1,86 @@
+import { useState } from 'react';
 import './App.css';
 import LikeNotActive from './img/like-not-active.svg';
-import { Link } from 'react-router-dom';
+import LikeActive from './img/like-active.svg';
+import { useNavigate } from 'react-router-dom';
 
-function Post({ img, likes, description, date, name, userAva, userId, setUserId }) {
+function Post({ postId, img, likes, description, date, name, userAva, userId, setUserId, userToken }) {
+    const [like, setLike] = useState(false);
+    const [likeState, setLikeState] = useState(likes);
+    const navigate = useNavigate();
+
+    const baseUrl = 'https://webdev-hw-api.vercel.app/api/v1/prod/instapro';
 
     const dateObj = new Date(date);
     const now = new Date();
     const differenceMs = now - dateObj;
     const differenceDays = Math.floor(differenceMs / (1000 * 60 * 60 * 24));
 
+    function onLikeClick() {
+        if (like) {
+            fetch(baseUrl + `/${postId}/dislike`, {
+                method: 'POST',
+                headers: {
+                    "Authorization": `Bearer ${userToken}`
+                }
+            })
+                .then(response => response.json())
+                .then(data => {
+                    data && setLike(false)
+                    refreshLikes()
+                })
+                .catch(error => console.log('ошибка:', error))
+        } else {
+            fetch(baseUrl + `/${postId}/like`, {
+                method: 'POST',
+                headers: {
+                    "Authorization": `Bearer ${userToken}`
+                }
+            })
+                .then(response => response.json())
+                .then(data => {
+                    data && setLike(true)
+                    refreshLikes()
+                })
+                .catch(error => console.log('ошибка:', error))

```

Comment:
ошибки лучше в error лог складывать 
```suggestion
                .catch(error => console.error(error))

```